components/paging/pager-load-more.js

```diff
@@ -0,0 +1,155 @@
+
+const nativeFocus = document.createElement('div').focus;
+
+/**
+ *  A pager component for load-more paging.
+ * @fires d2l-pager-load-more - Dispatched when the user clicks the load-more button. Consumers must call the provided "complete" method once items have been loaded.
+ */
+class LoadMore extends FocusMixin(FocusVisiblePolyfillMixin(LocalizeCoreElement(LitElement))) {
+
+	static get properties() {
+		return {
+			/**
+			 * Whether there are more items that can be loaded.
+			 * @type {boolean}
+			 */
+			hasMore: { type: Boolean, attribute: 'has-more', reflect: true },
+			/**
+			 * Total number of items. If not specified, neither it nor the count of items showing will be displayed.
+			 * @type {number}
+			 */
+			itemCount: { type: Number, attribute: 'item-count', reflect: true },
+			/**
+			 * The number of additional items to load.
+			 * @type {number}
+			 */
+			pageSize: { type: Number, attribute: 'page-size', reflect: true },
+			/**
+			 * The number of items showing. Assigned by PageableMixin.
+			 * @ignore
+			 */
+			itemShowingCount: { type: Number },
+			_loading: { state: true }
+		};
+	}
+
+	static get styles() {
+		return [ buttonStyles, labelStyles, css`
+			:host {
+				display: block;
+			}
+			:host(:not([has-more])),
+			:host([hidden]) {
+				display: none;
+			}
+			button {
+				align-items: center;
+				background-color: var(--d2l-color-regolith);
+				border: 1px solid var(--d2l-color-sylvite);
+				display: flex;
+				gap: 0.5rem;
+				justify-content: center;
+				width: 100%;
+			}
+			button:hover {
+				background-color: var(--d2l-color-sylvite);
+				border-color: var(--d2l-color-gypsum);
+			}
+			.action {
+				color: var(--d2l-color-celestine);
+			}
+			.separator {
+				border-right: 1px solid var(--d2l-color-mica);
+				height: 0.8rem;
+			}
+			.info {
+				color: var(--d2l-color-galena);
+				font-weight: 400;
+			}
+			d2l-loading-spinner {
+				display: none;
+			}
+			.loading > .action,
+			.loading > .separator,
+			.loading > .info {
+				display: none;
+			}
+			.loading > d2l-loading-spinner {
+				display: inline-block;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.hasMore = false;
+		this.itemCount = -1;
+		/** @ignore */
+		this.itemShowingCount = 0;
+		this.pageSize = 50;
+		this._loading = false;
+	}
+
+	static get focusElementSelector() {
+		return 'button';
+	}
+
+	render() {
+		if (!this.hasMore) return;
+		const classes = {
+			'd2l-label-text': true,
+			'loading': this._loading
+		};
+		return html`<button class="${classMap(classes)}" @click="${this._handleClick}">
+			<span class="action">${this.localize('components.pager-load-more.action', { count: this.pageSize })}</span>
+			${this.itemCount > -1 ? html`
+				<span class="separator"></span>
+				<span class="info">${this.localize('components.pager-load-more.info', { showingCount: this.itemShowingCount, totalCount: this.itemCount })}</span>
+			` : null}
```

Comment:
Nit: we've been using `nothing` in these cases as I believe it allows Lit to make some optimizations

Reply:
Yeah, it's probably good to get in the habit of this even if the result is the same.

In Lit 1, `null`, `undefined`, and an empty string all rendered an empty text node, which had the potential to suppress the default content of a `slot`. That was the primary use-case for using `nothing` then. (ex. `<my-element>${stuff}</my-element>` where stuff is `null | undefined` would suppress my-element's default slot content)

For Lit 2, the primary use-case now appears to be to normalize the binding behaviour for properties and attributes - `nothing` removes the attribute, whereas `null` and `undefined` render an empty string.  The Lit docs are slightly contradictory for [nothing](https://lit.dev/docs/api/templates/#nothing) and [Rendering Nothing](https://lit.dev/docs/templates/conditionals/#rendering-nothing) - `nothing` seems to be the more accurate of the two, as I've confirmed that that `null`, `undefined`, and empty string really do render no nodes in child contexts (the slot scenario above), so default slot content renders regardless of which value of those values is used.